libraries/shared/src/EntityItemID.cpp

```diff
@@ -0,0 +1,42 @@
+//
+//  EntityItemID.cpp
+//  libraries/shared/src
+//
+//  Created by Brad Hefta-Gaub on 12/4/13.
+//  Copyright 2013 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "EntityItemID.h"
+#include <QtCore/QObject>
+#include <QDebug>
+
+#include "BufferParser.h"
+#include "UUID.h"
+
+int entityItemIDTypeID = qRegisterMetaType<EntityItemID>();
+
+EntityItemID::EntityItemID() : QUuid()
+{
+}
+
+
+EntityItemID::EntityItemID(const QUuid& id) : QUuid(id)
+{
+}
+
+// EntityItemID::EntityItemID(const EntityItemID& other) : QUuid(other)
+// {
+// }
+
+EntityItemID EntityItemID::readEntityItemIDFromBuffer(const unsigned char* data, int bytesLeftToRead) {
+    EntityItemID result;
+    if (bytesLeftToRead >= NUM_BYTES_RFC4122_UUID) {
+        BufferParser(data, bytesLeftToRead).readUuid(result);
+    }
+    return result;
+}
+
+size_t std::hash<EntityItemID>::operator()(const EntityItemID& id) const { return qHash(id); }
```

Comment:
why'd this have to move to shared?

Reply:
I don't have strong proof right now, but I have memory that link-demand references to EntityItemID were infecting *every* library and I didn't want *everything* depending on script-engine, so I was able to remove the link-demand on script-engine from this object and moved it into shared.

Reply:
it doesn't seem like EntityItemID depends on script-engine anymore though, so maybe this can be moved back now?

Reply:
I'm more concerned about other stuff depending on EntityItemID and triggering a bunch of libraries to need to link to script-engine.

Reply:
oh I see what you mean...ok...